MapboxNavigation/PollyVoiceController.swift

```diff
@@ -144,24 +144,25 @@ public class PollyVoiceController: RouteVoiceController {
         if let audioPlayer = audioPlayer, audioPlayer.isPlaying, let lastSpokenInstruction = lastSpokenInstruction {
             voiceControllerDelegate?.voiceController?(self, didInterrupt: lastSpokenInstruction, with: instruction)
         }
+        let modifiedInstruction = voiceControllerDelegate?.voiceController?(self, willSpeak: instruction) ?? instruction
         pollyTask?.cancel()
         audioPlayer?.stop()
-        lastSpokenInstruction = instruction
+        lastSpokenInstruction = modifiedInstruction
         
-        guard spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString) == nil else {
-            play(spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString)! as Data)
+        guard spokenInstructionsForRoute.object(forKey: modifiedInstruction.ssmlText as NSString) == nil else {
```

Comment:
This may be better represented as: 

```swift
if let includedInstructions = spokenInstructionsForRoute.object(forKey: modifiedInstruction.ssmlText as NSString) {
    play(includedInstructions as Data)
    return
}
```